apps/tunes/management/commands/tunes_create_songs_from_spotify.py

```diff
@@ -14,76 +15,96 @@ def __init__(self, *args, **kwargs):
-            '''
-        )
-
     def save_songs_to_database(self, tracks):
         """
         Given a list of parameters for Song records, create the objects in the database.
-        :tracks: (list) List of dictionaries containing data to store for Song objects
-        @return: Two tuple of amount of tracks that successfully processed and how many failed to process
+
+        :param tracks: (list) List of dictionaries containing data to store for Song objects
+        :return: Two tuple of amount of tracks that successfully processed and how many failed to process
         """
         success, fail = 0, 0
         for track in tracks:
 
-            song, created = Song.objects.get_or_create(code=track['code'], defaults=track)
-
-            if created:
-                msg = 'Created song with code {}'.format(song.code)
-                self.write_to_log_and_output(msg)
-                success += 1
-            else:
-                msg = 'Song with code {} already exists'.format(song.code)
-                self.write_to_log_and_output(msg, output_stream='stderr')
+            try:
+                song, created = Song.objects.get_or_create(code=track['code'], defaults=track)
+
+                if created:
+                    msg = 'Created song with code {}'.format(song.code)
+                    self.write_to_log_and_output(msg)
+                    success += 1
+                else:
+                    msg = 'Song with code {} already exists'.format(song.code)
+                    self.write_to_log_and_output(msg)
+                    fail += 1
+            except ValidationError:
+                msg = 'ERROR: Could not create song with data: {}'.format(track)
+                self.write_to_log_and_output(msg, output_stream='stderr', log_level='warning')
                 fail += 1
 
         return success, fail
 
-    def handle(self, *args, **options):
-        self.logger.info('{} - Starting run to create songs from Spotify'.format(self._unique_id))
-
-        total_songs = options.get('total_songs')
-        num_playlists = 10
-        max_tracks_from_playlist = settings.SPOTIFY['max_songs_from_list']
-        max_tracks_from_category = settings.SPOTIFY['max_songs_from_category']
+    def get_tracks_from_spotify(self):
+        """
+        Request, format, and return tracks from Spotify's API.
 
+        :return: (list[dict]) Track data for saving as Song records
+        """
         spotify = SpotifyClient(command_id=self._unique_id)
-
         tracks = []
 
         for category in settings.SPOTIFY['categories']:
             songs_from_category = 0
 
-            playlists = spotify.get_playlists_for_category(category, num_playlists)
+            try:
+                playlists = spotify.get_playlists_for_category(category, settings.SPOTIFY['max_playlist_from_category'])
+                self.write_to_log_and_output('Got {} playlists for category: {}'.format(len(playlists), category))
```

Comment:
Perhaps we can just change this to a logging call? Seems superfluous to also write this to the console when it's a configurable setting